ibm/service/appconfiguration/data_source_ibm_app_config_segments.go

```diff
@@ -0,0 +1,473 @@
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Operator to be used for the evaluation if the entity belongs to the segment.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"next": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "URL to navigate to the next list of records.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"href": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "URL of the response.",
+						},
+					},
+				},
+			},
+			"first": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "URL to navigate to the first page of records.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"href": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "URL of the response.",
+						},
+					},
+				},
+			},
+			"previous": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "URL to navigate to the previous list of records.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"href": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "URL of the response.",
+						},
+					},
+				},
+			},
+			"last": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "URL to navigate to the last page of records.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"href": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "URL of the response.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIbmAppConfigSegmentsRead(d *schema.ResourceData, meta interface{}) error {
+	guid := d.Get("guid").(string)
+
+	appconfigClient, err := getAppConfigClient(meta, guid)
+	if err != nil {
+		return err
+	}
+
+	options := &appconfigurationv1.ListSegmentsOptions{}
+
+	if _, ok := d.GetOk("expand"); ok {
+		options.SetExpand(d.Get("expand").(bool))
+	}
+	if _, ok := d.GetOk("tags"); ok {
+		options.SetTags(d.Get("tags").(string))
+	}
+	if _, ok := d.GetOk("sort"); ok {
+		options.SetTags(d.Get("sort").(string))
+	}
+
+	if _, ok := d.GetOk("include"); ok {
+		options.SetInclude(d.Get("include").(string))
+	}
+
+	var segmentsList *appconfigurationv1.SegmentsList
+	var offset int64
+	var limit int64 = 10
+	var isLimit bool
+
+	finalList := []appconfigurationv1.Segment{}
+
+	if _, ok := d.GetOk("limit"); ok {
+		isLimit = true
+		limit = int64(d.Get("limit").(int))
+	}
+	options.SetLimit(limit)
+	if _, ok := d.GetOk("offset"); ok {
+		offset = int64(d.Get("offset").(int))
+	}
+	for {
+		options.Offset = &offset
+		result, response, err := appconfigClient.ListSegments(options)
+		segmentsList = result
+		if err != nil {
+			log.Printf("[DEBUG] ListSegments failed %s\n%s", err, response)
+			return err
```

Comment:
Kindly add formatting statements before returning error for better debugging experience..
`return fmt.Errof("ListSegments failed %s", err)`

Kindly make this change in all necessary places

Reply:
Addressed